src/libtiled/tilelayer.cpp

```diff
@@ -551,45 +592,56 @@ void TileLayer::offsetTiles(const QPoint &offset,
                             const QRect &bounds,
                             bool wrapX, bool wrapY)
 {
-    QVector<Cell> newGrid(mWidth * mHeight);
+    TileLayer *newLayer = clone();
 
-    for (int y = 0; y < mHeight; ++y) {
-        for (int x = 0; x < mWidth; ++x) {
-            // Skip out of bounds tiles
-            if (!bounds.contains(x, y)) {
-                newGrid[x + y * mWidth] = cellAt(x, y);
-                continue;
-            }
+    QHashIterator<QPoint, Chunk* > it(newLayer->mChunks);
+    while (it.hasNext()) {
+        it.next();
+        for (int y = 0; y < CHUNK_SIZE; ++y) {
+            for (int x = 0; x < CHUNK_SIZE; ++x) {
+                int _x = it.key().x() * CHUNK_SIZE + x;
+                int _y = it.key().y() * CHUNK_SIZE + y;
+
+                if (!contains(_x, _y))
+                    continue;
+
+                // Skip out of bounds tiles
+                if (!bounds.contains(_x, _y)) {
+                    newLayer->setCell(x, y, cellAt(_x, _y));
```

Comment:
You don't need to do this, since the `newLayer` is already a clone of the current layer. Actually, this means that you should only iterate the area indicated by `bounds`. Iterating just the chunks is not enough, since it could fail to include the needed area.